### PR TITLE
Enhancement #302 refactor notification system

### LINKED ISSUE #302: rename generic notification API
### Description

The name `MailerEvent` is actually invalid as the notification system aims to use the dispatcher to allow multiple endpoints where a notification can be published. Therefore it should be renamed to something like `NotificationEvent`. Furthermore we need a lightweight API which checks which channels (e.g. mailing) should be addressed and which parameters are necessary.

### PR DESCRIPTION
**Pull request by Ma27**
### Overview

This is a big refactoring scheduled before the first development release.
Ths dispatcher made things more fragmented and over-complicated the workflow (command handler -> notificatable trait -> notificator -> dispatcher -> concrete listener -> mailer).
Therefore the dispatcher has been removed in favor of a new, customized channel-based approach which configures channels using a custom DI tag. This gives the possibility to choose which channels can be used (so a user might be able to turn off email notifications for example). Furthermore a lot of code has been optimized and thanks to the removal of certain dependencies the complexity decreases.
#### Changed
- drop event dispatcher dependency
- minor code optimation
- support channels for each notifications
- improve DI tag configuration
### Related tickets

resolves #302

### CODE REVIEW

File: src/AppBundle/Service/Notification/Channel/MailingChannel.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+use AppBundle\Service\Notification\NotificationInput;
+use Swift_Mailer;
+use Symfony\Component\Templating\EngineInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * MailingChannel.
+ *
+ * @author Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ * @author Ben Bieler <benjaminbieler2014@gmail.com>
```
Comment (Ma27) [inline src/AppBundle/Service/Notification/Channel/MailingChannel.php:27]: @benbieler I keep you as author here as that's actually the old `MailListener` you built in #130, but GIT doesn't recognize this as moved file...

Comment (benbieler) [inline src/AppBundle/Service/Notification/Channel/MailingChannel.php:27]: Ok, fine

**Comment by Ma27** [comment]
@benbieler are you fine with it?

**Comment by benbieler** [comment]
@Ma27, seems fine to me, I do not see any obvious mistakes, but I have to mention, that since I am currently at work, I didn't have time to fully test it...

**Comment by Ma27** [comment]
ok so when the tests are green, we can merge this :D

**Pull request merged**